apps/dashboard/src/components/subscribers/subscriber-drawer.tsx

```diff
@@ -1,69 +1,48 @@
-import { motion } from 'motion/react';
-import { Sheet, SheetContentBase, SheetDescription, SheetPortal, SheetTitle } from '../primitives/sheet';
-import { VisuallyHidden } from '../primitives/visually-hidden';
-import { useNavigate, useParams } from 'react-router-dom';
-import { PropsWithChildren } from 'react';
-import { buildRoute, ROUTES } from '@/utils/routes';
+import { Sheet, SheetContent, SheetDescription, SheetTitle } from '@/components/primitives/sheet';
+import { VisuallyHidden } from '@/components/primitives/visually-hidden';
+import { SubscriberOverviewForm } from '@/components/subscribers/subscriber-overview-form';
+import { SubscriberOverviewSkeleton } from '@/components/subscribers/subscriber-overview-skeleton';
+import { useFetchSubscriber } from '@/hooks/use-fetch-subscriber';
+import { forwardRef, HTMLAttributes, useState } from 'react';
 
-const transitionSetting = { duration: 0.4 };
-
-type SubscriberDrawerProps = PropsWithChildren<{
+type SubscriberDrawerProps = {
   open: boolean;
-  onOpenChange?: (open: boolean) => void;
-}>;
+  onOpenChange: (open: boolean) => void;
+  subscriberId: string;
+};
 
-export function SubscriberDrawer({ children, open, onOpenChange }: SubscriberDrawerProps) {
-  const navigate = useNavigate();
-  const { environmentSlug } = useParams<{ environmentSlug: string }>();
+export const SubscriberDrawer = forwardRef<HTMLDivElement, SubscriberDrawerProps>((props, ref) => {
+  const { open, onOpenChange, subscriberId } = props;
 
-  const handleCloseSheet = () => {
-    navigate(
-      buildRoute(ROUTES.SUBSCRIBERS, {
-        environmentSlug: environmentSlug ?? '',
-      })
-    );
-  };
+  const { data: subscriber, isPending } = useFetchSubscriber({ subscriberId });
 
   return (
-    <Sheet modal={false} open={open} onOpenChange={onOpenChange}>
-      <motion.div
-        initial={{
-          opacity: 0,
-        }}
-        animate={{
-          opacity: 1,
-        }}
-        exit={{
-          opacity: 0,
+    <Sheet open={open} onOpenChange={onOpenChange}>
+      <SheetContent className="sm:max-w-lg" ref={ref}>
```

Comment:
it's missing the sidebar header:

![Screenshot 2025-02-07 at 18 12 26](https://github.com/user-attachments/assets/bb009e43-ca2d-4d56-b7e9-0e8461d56f50)

![Screenshot 2025-02-07 at 18 11 50](https://github.com/user-attachments/assets/33a87b5e-cb24-4f81-9ae0-f5e21f03f5c1)